TheWarWithin/DruidGuardian.lua

```diff
@@ -1051,6 +1055,12 @@ spec:RegisterHook( "runHandler", function( ability )
     end
 end )
 
+local ursocRageSpend
```

Comment:
This is probably not working as intended, unless I've overlooked something.

You need a "real" value that shows how much Ursoc-rage has actually been spent (probably from UNIT_POWER_FREQUENT), then you need a "virtual" value that updates during the prediction period.

(This is assuming there's no API that will track the Rage expenditure.)

As written, this will (1) reset on combat and (2) increase with every Rage spender recommendation for every set of recommendations *even if you sit idle*.

Example from WarriorFury.lua

```Lua
local RAGE = Enum.PowerType.Rage
local lastRage = -1
local rageSpent = 0

spec:RegisterUnitEvent( "UNIT_POWER_FREQUENT", "player", nil, function( event, unit, powerType )
    if powerType == "RAGE" then
        local current = UnitPower( "player", RAGE )
        if current < lastRage - 3 then -- Spent Rage, -3 is used as a Hack to avoid Rage decay triggering
            if state.talent.anger_management.enabled then
                rageSpent = ( rageSpent + ( lastRage - current ) ) % 20
            end
        end
        lastRage = current
    end
end )

spec:RegisterStateExpr( "rage_spent", function ()
    return rageSpent
end )
```

You would also want to reset `rageSpent` (real value) on `PLAYER_REGEN_ENABLED` or `PLAYER_REGEN_DISABLED` events, so the real value resets as you set up.

Then, in `reset_precast` you need to reset `rage_spent` to `nil` (so it pulls from the real `rageSpent` value initially for each set of recommendations).  And you manipulate `rage_spent` in the prediction space (hooks, handlers, etc.).

```Lua
        if talent.ursocs_guidance.enabled then
            rage_spent = rage_spent + amt
            if rage_spent >= 25 then
                reduceCooldown( "incarnation", floor( rage_spent / 25 ) )
                rage_spent = rage_spent % 25
            end
        end
```

Reply:
(Just for general awareness, if something is camel case, it generally assumes that it's not intended for use in a priority, while following SimC's token (`abc_def`) format is for stuff that could be used and is sanitized in resets/updates.